genrl/bandit/agents/cb_agents/common/bayesian.py

```diff
@@ -0,0 +1,198 @@
+
+
+class BayesianLinear(nn.Module):
+    """Linear Layer for Bayesian Neural Networks.
+
+    Args:
+        in_features (int): size of each input sample
+        out_features (int): size of each output sample
+        bias (bool, optional): Whether to use an additive bias. Defaults to True.
+    """
+
+    def __init__(self, in_features: int, out_features: int, bias: bool = True) -> None:
+        super(BayesianLinear, self).__init__()
+
+        self.in_features = in_features
+        self.out_features = out_features
+        self.bias = bias
+
+        self.w_mu = nn.Parameter(torch.Tensor(out_features, in_features))
+        self.w_sigma = nn.Parameter(torch.Tensor(out_features, in_features))
+        if self.bias:
+            self.b_mu = nn.Parameter(torch.Tensor(out_features))
+            self.b_sigma = nn.Parameter(torch.Tensor(out_features))
+
+        self.reset_parameters()
+
+    def reset_parameters(self) -> None:
+        """Resets weight and bias parameters of the layer.
+        """
+        self.w_mu.data.normal_(0, 0.1)
+        self.w_sigma.data.normal_(0, 0.1)
+        if self.bias:
+            self.b_mu.data.normal_(0, 0.1)
+            self.b_sigma.data.normal_(0, 0.1)
+
+    def forward(
+        self, x: torch.Tensor, kl: bool = True, frozen: bool = False
+    ) -> Tuple[torch.Tensor, Optional[torch.Tensor]]:
+        """Apply linear transormation to input.
+
+        The weight and bias is sampled for each forward pass from a normal
+        distribution. The KL divergence of the sampled weigth and bias can
+        also be computed if specified.
+
+        Args:
+            x (torch.Tensor): Input to be transformed
+            kl (bool, optional): Whether to compute the KL divergence. Defaults to True.
+            frozen (bool, optional): Whether to freeze current parameters. Defaults to False.
+
+        Returns:
+            Tuple[torch.Tensor, Optional[torch.Tensor]]: The transformed input and optionally
+                the computed KL divergence value.
+        """
+        kl_val = None
+        b = None
+        if frozen:
+            w = self.w_mu
+            if self.bias:
+                b = self.b_mu
+        else:
+            w_dist = torch.distributions.Normal(self.w_mu, self.w_sigma)
+            w = w_dist.rsample()
+            if kl:
+                kl_val = torch.sum(
+                    w_dist.log_prob(w) - torch.distributions.Normal(0, 0.1).log_prob(w)
+                )
+            if self.bias:
+                b_dist = torch.distributions.Normal(self.b_mu, self.b_sigma)
+                b = b_dist.rsample()
+                if kl:
+                    kl_val += torch.sum(
+                        b_dist.log_prob(b)
+                        - torch.distributions.Normal(0, 0.1).log_prob(b)
+                    )
+            else:
+                b = 0.0
+                # b_logprob = None
+
+        return F.linear(x, w, b), kl_val
```

Comment:
The codeclimate issue can be resolved by putting in as few if else statements as possible. This is a pretty long block of code with lots of nested if statements. Would be better if you could cover this in fewer lines with lesser if statements. (atleast try to go without nested). Inline if else assignments may help too.